public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -20,21 +14,10 @@ export class TablePanel extends Component<Props> {
     super(props);
   }
 
-  onColumnResize = (fieldIndex: number, width: number) => {
-    const { fieldConfig, data } = this.props;
+  onColumnResize = (fieldDisplayName: string, width: number) => {
```

Comment:
name may not be unique -- we should never use it as a key

Reply:
Do we have an alternative?

Reply:
if it were just transient - the fieldIndex would be better... but if we persist the values then name makes more sense.  Either way they will each have weird behaviors for things with dupilcate names.

But the override configs UI would look better if they are by name, so I think you are right

Reply:
hmm, was thinking the same but at the same time we are using it as the identifier in transforms and on other places.

Is it possible to combine a natural ID in some way? Like fieldName + some kind of query identifier?

Reply:
Maybe I am totally offside now but would the displayName + query refId work? I am guessing that the refId is identifying the actual query and not the execution of the query.